### ISSUE TITLE
Support for Python 3.8.7

### ISSUE THREAD

**Issue by rafaelvp-db** [databrickslabs/dbx#328]
## Expected Behavior

Possible to install `dbx` with Python 3.8.7 without any issues

## Current Behavior

Installation fails due to scipy dependency version.

When installing with Python 3.7.13 it works without any issues. However main implication is that some of the code using more recent versions of `scikit-learn` (e.g. 1.1.1) stops working. This could be the case for other libraries as well.

## Steps to Reproduce (for bugs)

1. Install pyenv
2. Install Python 3.8.7
3. `pip install dbx`

## Your Environment

* dbx version used: latest
* Databricks Runtime version: NA

**Comment by rafaelvp-db** [comment]
hey @renardeinside, thanks for the feedback! indeed, I don't see scipy as a direct dependency of the package but it might be installed by one of the requirements. please find the logs below:

```pip install dbx
Collecting dbx
  Using cached dbx-0.6.4-py3-none-any.whl (80 kB)
Collecting pydantic>=1.9.1
  Using cached pydantic-1.9.1-py3-none-any.whl (142 kB)
Collecting Jinja2>=2.11.2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting tqdm>=4.50.0
  Using cached tqdm-4.64.0-py2.py3-none-any.whl (78 kB)
Collecting retry>=0.9.2
  Using cached retry-0.9.2-py2.py3-none-any.whl (8.0 kB)
Processing /Users/rafael.pierre/Library/Caches/pip/wheels/5e/8c/80/c3646df8201ba6f5070297fe3779a4b70265d0bfd961c15302/emoji-1.7.0-py3-none-any.whl
Collecting click>=7.1.2
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting mlflow>=1.26.1
  Using cached mlflow-1.27.0-py3-none-any.whl (17.9 MB)
Collecting azure-mgmt-subscription>=3.0.0
  Using cached azure_mgmt_subscription-3.0.0-py3-none-any.whl (57 kB)
Collecting aiohttp>=3.8.1
  Using cached aiohttp-3.8.1.tar.gz (7.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
Collecting pathspec>=0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting azure-mgmt-datafactory>=2.2.0
  Using cached azure_mgmt_datafactory-2.7.0-py3-none-any.whl (435 kB)
Collecting databricks-cli>=0.16.4
  Using cached databricks-cli-0.17.0.tar.gz (81 kB)
Processing /Users/rafael.pierre/Library/Caches/pip/wheels/52/84/66/50912fd7bf1639a31758e40bd4312602e104a8eca1e0da9645/PyYAML-6.0-cp38-cp38-macosx_12_0_x86_64.whl
Collecting requests>=2.24.0
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting cryptography<38.0.0,>=3.3.1
  Using cached cryptography-37.0.4.tar.gz (585 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting watchdog>=2.1.0
  Using cached watchdog-2.1.9.tar.gz (119 kB)
Collecting azure-identity>=1.7.1
  Using cached azure_identity-1.10.0-py3-none-any.whl (134 kB)
Collecting cookiecutter>=1.7.2
  Using cached cookiecutter-2.1.1-py2.py3-none-any.whl (36 kB)
Collecting typing-extensions>=3.7.4.3
  Using cached typing_extensions-4.3.0-py3-none-any.whl (25 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1.tar.gz (18 kB)
Collecting decorator>=3.4.2
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting py<2.0.0,>=1.4.26
  Using cached py-1.11.0-py2.py3-none-any.whl (98 kB)
Collecting querystring-parser
  Using cached querystring_parser-1.2.4-py2.py3-none-any.whl (7.9 kB)
Collecting entrypoints
  Using cached entrypoints-0.4-py3-none-any.whl (5.3 kB)
Collecting alembic
  Downloading alembic-1.8.1-py3-none-any.whl (209 kB)
     |████████████████████████████████| 209 kB 3.7 MB/s 
Collecting sqlparse>=0.3.1
  Using cached sqlparse-0.4.2-py3-none-any.whl (42 kB)
Collecting sqlalchemy>=1.4.0
  Using cached SQLAlchemy-1.4.39.tar.gz (8.2 MB)
Collecting gitpython>=2.1.0
  Using cached GitPython-3.1.27-py3-none-any.whl (181 kB)
Collecting gunicorn; platform_system != "Windows"
  Using cached gunicorn-20.1.0-py3-none-any.whl (79 kB)
Collecting prometheus-flask-exporter
  Using cached prometheus_flask_exporter-0.20.2-py3-none-any.whl (18 kB)
Collecting Flask
  Downloading Flask-2.1.3-py3-none-any.whl (95 kB)
     |████████████████████████████████| 95 kB 14.2 MB/s 
Collecting protobuf>=3.12.0
  Using cached protobuf-4.21.2-py2.py3-none-any.whl (164 kB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting numpy
  Downloading numpy-1.23.1.tar.gz (10.7 MB)
     |████████████████████████████████| 10.7 MB 9.1 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting cloudpickle
  Using cached cloudpickle-2.1.0-py3-none-any.whl (25 kB)
Collecting docker>=4.0.0
  Using cached docker-5.0.3-py2.py3-none-any.whl (146 kB)
Collecting scipy
  Using cached scipy-1.8.1.tar.gz (38.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/rafael.pierre/projects/dbx-scipy/.venv/bin/python /Users/rafael.pierre/projects/dbx-scipy/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/tmpt4rsjfkg
         cwd: /private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-install-qvy9kshh/scipy
    Complete output (218 lines):
    Running scipy/linalg/_generate_pyx.py
    Running scipy/special/_generate_pyx.py
    Running scipy/stats/_generate_pyx.py
    Processing scipy/cluster/_vq.pyx
    Processing scipy/cluster/_optimal_leaf_ordering.pyx
    Processing scipy/cluster/_hierarchy.pyx
    Processing scipy/ndimage/src/_cytest.pyx
    Processing scipy/ndimage/src/_ni_label.pyx
    Processing scipy/linalg/cython_lapack.pyx
    Processing scipy/linalg/_matfuncs_sqrtm_triu.pyx
    Processing scipy/linalg/_cythonized_array_utils.pyx
    Processing scipy/linalg/_solve_toeplitz.pyx
    Processing scipy/linalg/cython_blas.pyx
    Processing scipy/linalg/_decomp_update.pyx.in
    Processing scipy/optimize/_group_columns.pyx
    Processing scipy/optimize/_bglu_dense.pyx
    Processing scipy/optimize/_trlib/_trlib.pyx
    Processing scipy/optimize/_highs/cython/src/_highs_constants.pyx
    Processing scipy/optimize/_highs/cython/src/_highs_wrapper.pyx
    Processing scipy/optimize/_lsq/givens_elimination.pyx
    Processing scipy/optimize/cython_optimize/_zeros.pyx.in
    Processing scipy/optimize/tnc/_moduleTNC.pyx
    Processing scipy/io/matlab/_streams.pyx
    Processing scipy/io/matlab/_mio_utils.pyx
    Processing scipy/io/matlab/_mio5_utils.pyx
    Processing scipy/_lib/_ccallback_c.pyx
    Processing scipy/_lib/_test_deprecation_def.pyx
    Processing scipy/_lib/_test_deprecation_call.pyx
    Processing scipy/_lib/messagestream.pyx
    Processing scipy/special/_ufuncs_cxx.pyx
    Processing scipy/special/cython_special.pyx
    Processing scipy/special/_ellip_harm_2.pyx
    Processing scipy/special/_comb.pyx
    Processing scipy/special/_test_round.pyx
    Processing scipy/special/_ufuncs.pyx
    Processing scipy/fftpack/convolve.pyx
    Processing scipy/interpolate/interpnd.pyx
    Processing scipy/interpolate/_bspl.pyx
    Processing scipy/interpolate/_ppoly.pyx
    Processing scipy/sparse/_csparsetools.pyx.in
    Processing scipy/sparse/csgraph/_shortest_path.pyx
    Processing scipy/sparse/csgraph/_traversal.pyx
    Processing scipy/sparse/csgraph/_flow.pyx
    Processing scipy/sparse/csgraph/_tools.pyx
    Processing scipy/sparse/csgraph/_matching.pyx
    Processing scipy/sparse/csgraph/_reordering.pyx
    Processing scipy/sparse/csgraph/_min_spanning_tree.pyx
    Processing scipy/spatial/_qhull.pyx
    Processing scipy/spatial/_ckdtree.pyx
    Processing scipy/spatial/_voronoi.pyx
    Processing scipy/spatial/_hausdorff.pyx
    Processing scipy/spatial/transform/_rotation.pyx
    Processing scipy/signal/_max_len_seq_inner.pyx
    Processing scipy/signal/_peak_finding_utils.pyx
    Processing scipy/signal/_upfirdn_apply.pyx
    Processing scipy/signal/_spectral.pyx
    Processing scipy/signal/_sosfilt.pyx
    Processing scipy/stats/_stats.pyx
    Processing scipy/stats/_qmc_cy.pyx
    Processing scipy/stats/_biasedurn.pyx
    Processing scipy/stats/_sobol.pyx
    Processing scipy/stats/_boost/src/beta_ufunc.pyx
    Processing scipy/stats/_boost/src/nbinom_ufunc.pyx
    Processing scipy/stats/_boost/src/hypergeom_ufunc.pyx
    Processing scipy/stats/_boost/src/binom_ufunc.pyx
    Processing scipy/stats/_unuran/unuran_wrapper.pyx
    warning: unuran_wrapper.pyx:470:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:470:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:470:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    Cythonizing sources
    lapack_opt_info:
    lapack_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/Users/rafael.pierre/projects/dbx-scipy/.venv/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    openblas_lapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas not found in ['/Users/rafael.pierre/projects/dbx-scipy/.venv/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    openblas_clapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas,lapack not found in ['/Users/rafael.pierre/projects/dbx-scipy/.venv/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    flame_info:
    customize UnixCCompiler
      libraries flame not found in ['/Users/rafael.pierre/projects/dbx-scipy/.venv/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    atlas_3_10_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
      NOT AVAILABLE
    
    atlas_3_10_info:
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_info'>
      NOT AVAILABLE
    
    atlas_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_threads_info'>
      NOT AVAILABLE
    
    atlas_info:
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /Users/rafael.pierre/projects/dbx-scipy/.venv/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_info'>
      NOT AVAILABLE
    
    accelerate_info:
      NOT AVAILABLE
    
    lapack_info:
    customize UnixCCompiler
      libraries lapack not found in ['/Users/rafael.pierre/projects/dbx-scipy/.venv/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    lapack_src_info:
      NOT AVAILABLE
    
      NOT AVAILABLE
    
    setup.py:486: UserWarning: Unrecognized setuptools command ('dist_info --egg-base /private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-modern-metadata-wgo3moj_'), proceeding with generating Cython sources and expanding templates
      warnings.warn("Unrecognized setuptools command ('{}'), proceeding with "
    Running from SciPy source directory.
    /private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/numpy/distutils/system_info.py:1712: UserWarning:
        Lapack (http://www.netlib.org/lapack/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [lapack]) or by setting
        the LAPACK environment variable.
      if getattr(self, '_calc_info_{}'.format(lapack))():
    /private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/numpy/distutils/system_info.py:1712: UserWarning:
        Lapack (http://www.netlib.org/lapack/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [lapack_src]) or by setting
        the LAPACK_SRC environment variable.
      if getattr(self, '_calc_info_{}'.format(lapack))():
    Traceback (most recent call last):
      File "/Users/rafael.pierre/projects/dbx-scipy/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
        main()
      File "/Users/rafael.pierre/projects/dbx-scipy/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/Users/rafael.pierre/projects/dbx-scipy/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 133, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 174, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 267, in run_setup
        super(_BuildMetaLegacyBackend,
      File "/private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 158, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 628, in <module>
        setup_package()
      File "setup.py", line 624, in setup_package
        setup(**metadata)
      File "/private/var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/pip-build-env-izffbc6j/overlay/lib/python3.8/site-packages/numpy/distutils/core.py", line 137, in setup
        config = configuration()
      File "setup.py", line 526, in configuration
        raise NotFoundError(msg)
    numpy.distutils.system_info.NotFoundError: No BLAS/LAPACK libraries found. Note: Accelerate is no longer supported.
    To build Scipy from sources, BLAS & LAPACK libraries need to be installed.
    See site.cfg.example in the Scipy source directory and
    https://docs.scipy.org/doc/scipy/reference/building/index.html for details.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/rafael.pierre/projects/dbx-scipy/.venv/bin/python /Users/rafael.pierre/projects/dbx-scipy/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /var/folders/np/w2_t05c16912f_0jvc4rj64r0000gp/T/tmpt4rsjfkg Check the logs for full command output.```

Python Version: 3.8.7

**Comment by rafaelvp-db** [comment]
all right, so scipy is a requirement for installing mlflow. there's a known issue with installing scipy for some of the python 3.8.x versions - as there's no wheel available for these versions, pip tries to build scipy from the source. I tried to install the dependencies (openblas etc), without success. but when I followed the instructions [from this link](https://stackoverflow.com/questions/71006857/macos-monterey-install-scipy-error-no-lapack-blas-resources-found), I was able to download the wheel for scipy and the problem was solved.
